base/dict.jl

```diff
@@ -696,33 +695,40 @@ function skip_deleted(h::Dict, i)
     end
     return 0
 end
-function skip_deleted_floor!(h::Dict)
-    idx = skip_deleted(h, h.idxfloor)
-    if idx != 0
-        h.idxfloor = idx
-    end
-    idx
-end
 
-@propagate_inbounds _iterate(t::Dict{K,V}, i) where {K,V} = i == 0 ? nothing : (Pair{K,V}(t.keys[i],t.vals[i]), i == typemax(Int) ? 0 : i+1)
+
 @propagate_inbounds function iterate(t::Dict)
-    _iterate(t, skip_deleted(t, t.idxfloor))
+    isempty(t) && return nothing
+    _iterate(t, t.count==0 ? 0 : skip_deleted(t, 1))
+end
+@propagate_inbounds function _iterate(t::Dict{K,V}, i) where {K,V}
+    # overflow check not needed on i+1 because a dict that large would more than fill memory
+    return (Pair{K,V}(t.keys[i],t.vals[i]), i+1)
```

Comment:
```suggestion
    return i == 0 ? nothing : (Pair{K,V}(t.keys[i],t.vals[i]), i+1)
```
We need to revert this to fix the bootstraping.

Reply:
Friendly bumb @oscardssmith